home/screens/HomeScreen/index.tsx

```diff
@@ -0,0 +1,277 @@
+import { spacing } from '@expo/styleguide-native';
+import { StackScreenProps } from '@react-navigation/stack';
+import { HomeScreenHeader } from 'components/HomeScreenHeader';
+import Constants from 'expo-constants';
+import { View, Divider, ThemeContext } from 'expo-dev-client-components';
+import { CurrentUserDataFragment, useHome_CurrentUserQuery } from 'graphql/types';
+import * as React from 'react';
+import { Alert, AppState, Platform, StyleSheet } from 'react-native';
+import { SafeAreaView } from 'react-native-safe-area-context';
+
+import ApiV2HttpClient from '../../api/ApiV2HttpClient';
+import Connectivity from '../../api/Connectivity';
+import ScrollView from '../../components/NavigationScrollView';
+import RefreshControl from '../../components/RefreshControl';
+import ThemedStatusBar from '../../components/ThemedStatusBar';
+import { HomeStackRoutes } from '../../navigation/Navigation.types';
+import { useDispatch, useSelector } from '../../redux/Hooks';
+import { DevSession, HistoryList } from '../../types';
+import addListenerWithNativeCallback from '../../utils/addListenerWithNativeCallback';
+import getSnackId from '../../utils/getSnackId';
+import isUserAuthenticated from '../../utils/isUserAuthenticated';
+import { DevelopmentServerListItem } from './DevelopmentServerListItem';
+import { DevelopmentServersHeader } from './DevelopmentServersHeader';
+import { DevelopmentServersPlaceholder } from './DevelopmentServersPlaceholder';
+
+const PROJECT_UPDATE_INTERVAL = 10000;
+
+type Props = NavigationProps & {
+  dispatch: (data: any) => any;
+  isFocused: boolean;
+  recentHistory: HistoryList;
+  allHistory: HistoryList;
+  isAuthenticated: boolean;
+  currentUser?: CurrentUserDataFragment;
+  theme: string;
+};
+
+type State = {
+  projects: DevSession[];
+  isNetworkAvailable: boolean;
+  isRefreshing: boolean;
+};
+
+type NavigationProps = StackScreenProps<HomeStackRoutes, 'Home'>;
+
+export default function HomeScreen(props: NavigationProps) {
+  const [isFocused, setFocused] = React.useState(true);
+  React.useEffect(() => {
```

Comment:
did the useIsFocus hook not work here? https://reactnavigation.org/docs/use-is-focused/